tempest/api/workloadmgr/security_groups/test_security_group_default.py

```
@@ -0,0 +1,322 @@
+                try:
+                    LOG.debug("Delete VM + volume + security groups")
+                    self.delete_vm_secgroups([self.vm_id], created_security_groups)
+                    reporting.add_test_step(
+                        "Delete vm and security group", tvaultconf.PASS
+                    )
+                except Exception as e:
+                    LOG.error(f"Exception: {e}")
+                    raise Exception("Deletion of vm and security group/s failed")
+                time.sleep(60)
+
+                # Perform security groups restore through wlm command
+                self._restore_through_wlm_cli(snapshot_id)
+
+                # Security group and rules verification post restore
+                LOG.debug("Compare the security groups before and after restore")
+                secgroups_after = self.list_security_groups()
+                if len(secgroups_after) == len(secgroups_before):
+                    reporting.add_test_step(
+                        "Security group verification pre and post restore",
+                        tvaultconf.PASS,
+                    )
+                else:
+                    LOG.error("Security groups verification pre and post restore")
+                    reporting.add_test_step(
+                        "Security groups verification failed pre and post restore",
+                        tvaultconf.FAIL,
+                    )
+
+                LOG.debug("Compare the security group rules before and after restore")
+                rules_after = self.list_security_group_rules()
+                if len(rules_after) == len(rules_before):
+                    reporting.add_test_step(
+                        "Security group rules verification pre and post restore",
+                        tvaultconf.PASS,
+                    )
+                else:
+                    reporting.add_test_step(
+                        "Security group rules verification pre and post restore",
```

test script status also need to be marked as FAIL explicitly

---

done